spec/machine/AARCH64/Arch_Kernel_Config_Lemmas.thy

```diff
@@ -117,5 +117,15 @@ lemma maxIRQ_1_plus_eq_Suc_machine[simp]:
   "unat (1 + Kernel_Config.maxIRQ :: machine_word) = Suc Kernel_Config.maxIRQ"
   by (simp add: Kernel_Config.maxIRQ_def)
 
+
+(* cacheLineBits conditions *)
+
+(* Folding cacheLineBits_val in C functions only works reliably if cacheLineBits is not 1 and
+   not too large to conflict with other values used inside cache ops.
+   12 is ptBits, which is only available after ExecSpec. Anything smaller than ptBits works. *)
```

Comment:
This is a bit of a worry, since it's a purely syntactic restriction. What is the risk that the proofs will fail in the future due to the appearance of some other number that happens to be the same as cacheLineBits?

Reply:
The risk is very low. This code hasn't changed in a very long time, and it would be fairly strange to mention other numbers in these functions that happen to be in that cache line size range. The only other literal number that occurs in these functions before simplification is `1` (for loop counter increment and a few other expressions).

Reply:
(the upper range of the interval comes from the proof constraints)